tests/algorithms/test_gradient_clipping.py

```diff
@@ -192,12 +193,27 @@ def test_gradient_clipping_algorithm_with_deepspeed_enabled(
     apply_gc_fn.assert_not_called()
 
 
-def _auto_wrap_policy(module: torch.nn.Module, recurse: bool, unwrapped_params: int) -> bool:
-    if recurse:
-        return True
-    if hasattr(module, '_fsdp_wrap'):
-        return bool(module._fsdp_wrap)
-    return False
+if not using_torch_2_0():
+
+    def _auto_wrap_policy(module: torch.nn.Module, recurse: bool, unwrapped_params: int) -> bool:  # type: ignore
+        if recurse:
+            return True
+        if hasattr(module, '_fsdp_wrap'):
+            return bool(module._fsdp_wrap)
+        return False
+else:
+
+    def _auto_wrap_policy(module: torch.nn.Module, recurse: bool, nonwrapped_numel: int) -> bool:
+        if recurse:
+            return True
+
+        # With Torch 2.0, there is a bug that emits a nasty warning if you wrap a module with no parameters
+        if len(list(module.parameters())) == 0:
+            return False
```

Comment:
Do we need to special case this w 2.0 or can we always do this?

Reply:
Unfortunately there is also a parameter renamed between these two functions.